### PR TITLE
Implemented Question Controller

### PR DESCRIPTION
**Pull request by DhanashreeChandgude**
Hi @Akkshay3,
Please review my code changes for The following API endpoints

1. createQuestion - "/question/create"
2. getAllQuestions - "/question/all"
3. editQuestionContent - "/question/edit/{questionId}"
4. deleteQuestion - "/question/delete/{questionId}"
5. getAllQuestionsByUser - "question/all/{userId}"

Please review and merge.

### CODE REVIEW

File: quora-api/src/main/java/com/upgrad/quora/api/controller/QuestionController.java [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+    /**
+     * Controller method to handle createQuestion POST endpoint
+     *
+     * @param questionRequest
+     * @param authorization
+     * @return QuestionResponse
+     * @throws AuthorizationFailedException
+     */
+    @RequestMapping(
```
Comment (Akkshay3) [inline quora-api/src/main/java/com/upgrad/quora/api/controller/QuestionController.java]: Use @PostMapping

File: quora-api/src/main/java/com/upgrad/quora/api/controller/QuestionController.java [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    }
+
+    /**
+     * Controller method to handle GET request to fetch all questions
+     *
+     * @param authorization
+     * @return List of all questions
+     * @throws AuthorizationFailedException
+     */
+    @RequestMapping(
```
Comment (Akkshay3) [inline quora-api/src/main/java/com/upgrad/quora/api/controller/QuestionController.java]: Use @GetMapping

File: quora-api/src/main/java/com/upgrad/quora/api/controller/QuestionController.java [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+     * Controller method to handle PUT request to update question
+     *
+     * @param questionEditRequest
+     * @param questionUuid
+     * @param authorization
+     * @return QuestionEditResponse
+     * @throws AuthorizationFailedException
+     * @throws InvalidQuestionException
+     */
+    @RequestMapping(
```
Comment (Akkshay3) [inline quora-api/src/main/java/com/upgrad/quora/api/controller/QuestionController.java]: Use @PutMapping

File: quora-api/src/main/java/com/upgrad/quora/api/controller/QuestionController.java [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+            method = RequestMethod.PUT,
+            path = "/question/edit/{questionId}",
+            consumes = MediaType.APPLICATION_JSON_UTF8_VALUE,
```
Comment (Akkshay3) [inline quora-api/src/main/java/com/upgrad/quora/api/controller/QuestionController.java]: Remove mediatype information as default is JSON

File: quora-api/src/main/java/com/upgrad/quora/api/controller/QuestionController.java [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+    /**
+     * Controller method to get all questions posted by a user
+     *
+     * @param accessToken
+     * @param userId
+     * @return
+     * @throws AuthorizationFailedException
+     * @throws UserNotFoundException
+     */
+    @RequestMapping(
```
Comment (Akkshay3) [inline quora-api/src/main/java/com/upgrad/quora/api/controller/QuestionController.java]: change to @GetMapping

File: quora-api/src/main/java/com/upgrad/quora/api/controller/QuestionController.java [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+    /**
+     * Controller method to handle DELETE request to delete question
+     *
+     * @param accessToken
+     * @param questionId
+     * @return
+     * @throws AuthorizationFailedException
+     * @throws InvalidQuestionException
+     */
+    @RequestMapping(method = RequestMethod.DELETE, path = "/question/delete/{questionId}")
```
Comment (Akkshay3) [inline quora-api/src/main/java/com/upgrad/quora/api/controller/QuestionController.java]: change to @DeleteMapping

File: quora-api/src/main/java/com/upgrad/quora/api/exception/QuoraExceptionHandler.java [hunk 7]
```diff
@@ -6,9 +6,10 @@
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.ControllerAdvice;
 import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
 import org.springframework.web.context.request.WebRequest;
 
-@ControllerAdvice
+@RestControllerAdvice
```
Comment (Akkshay3) [inline quora-api/src/main/java/com/upgrad/quora/api/exception/QuoraExceptionHandler.java:11]: Nice

File: quora-service/src/main/java/com/upgrad/quora/service/dao/QuestionRepository.java [hunk 8]
```diff
@@ -0,0 +1,12 @@
+package com.upgrad.quora.service.dao;
+
+import com.upgrad.quora.service.entity.QuestionEntity;
+import com.upgrad.quora.service.entity.UserEntity;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import java.util.List;
+
+public interface QuestionRepository  extends JpaRepository<QuestionEntity, Integer> {
```
Comment (Akkshay3) [inline quora-service/src/main/java/com/upgrad/quora/service/dao/QuestionRepository.java:12]: Add java doc

File: quora-api/src/main/java/com/upgrad/quora/api/controller/QuestionController.java (hunk 3 above)
Comment (DhanashreeChandgude) [reply quora-api/src/main/java/com/upgrad/quora/api/controller/QuestionController.java]: Done

**Review by Akkshay3 — APPROVED** [review]
Approved!

**Pull request merged**